Content.Server/_White/WeaponModules/WeaponModulesSystem.cs

```diff
@@ -0,0 +1,226 @@
+﻿using System.Diagnostics.CodeAnalysis;
+using Content.Shared._White.WeaponModules;
+using Content.Shared.Weapons.Ranged.Components;
+using Content.Shared.Weapons.Ranged.Systems;
+using Robust.Server.GameObjects;
+using Robust.Shared.Containers;
+
+namespace Content.Server._White.WeaponModules;
+
+public sealed class WeaponModulesSystem : EntitySystem
+{
+    protected const string ModulesSlot = "gun_modules";
+    [Dependency] private readonly PointLightSystem _lightSystem = default!;
+    [Dependency] private readonly SharedAppearanceSystem _appearanceSystem = default!;
+    [Dependency] private readonly SharedGunSystem _gunSystem = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<LightModuleComponent, EntGotInsertedIntoContainerMessage>(LightModuleOnInsert);
+        SubscribeLocalEvent<LightModuleComponent, EntGotRemovedFromContainerMessage>(LightModuleOnEject);
+
+        SubscribeLocalEvent<LaserModuleComponent, EntGotInsertedIntoContainerMessage>(LaserModuleOnInsert);
+        SubscribeLocalEvent<LaserModuleComponent, EntGotRemovedFromContainerMessage>(LaserModuleOnEject);
+
+        SubscribeLocalEvent<FlameHiderModuleComponent, EntGotInsertedIntoContainerMessage>(FlameHiderModuleOnInsert);
+        SubscribeLocalEvent<FlameHiderModuleComponent, EntGotRemovedFromContainerMessage>(FlameHiderModuleOnEject);
+
+        SubscribeLocalEvent<SilencerModuleComponent, EntGotInsertedIntoContainerMessage>(SilencerModuleOnInsert);
+        SubscribeLocalEvent<SilencerModuleComponent, EntGotRemovedFromContainerMessage>(SilencerModuleOnEject);
+
+        SubscribeLocalEvent<AcceleratorModuleComponent, EntGotInsertedIntoContainerMessage>(AcceleratorModuleOnInsert);
+        SubscribeLocalEvent<AcceleratorModuleComponent, EntGotRemovedFromContainerMessage>(AcceleratorModuleOnEject);
+    }
+    private bool TryInsertModule(EntityUid module, EntityUid weapon, EntGotInsertedIntoContainerMessage args, [NotNullWhen(true)] out WeaponModulesComponent? weaponModulesComponent)
+    {
+        if (TryComp<WeaponModulesComponent>(weapon, out var weaponModulesComponents) && ModulesSlot == args.Container.ID)
+        {
+            if(!weaponModulesComponents.Modules.Contains(module))
+                weaponModulesComponents.Modules.Add(module);
+            weaponModulesComponent = weaponModulesComponents;
+            return true;
+        }
+        weaponModulesComponent = weaponModulesComponents;
+        return false;
+    }
```

Comment:
```suggestion
    private bool TryInsertModule(EntityUid module, EntityUid weapon, BaseModuleComponent component,
        EntGotInsertedIntoContainerMessage args, [NotNullWhen(true)] out WeaponModulesComponent? weaponModulesComponent,
        [NotNullWhen(true)] out AppearanceComponent? appearanceComponent)
    {
        if (!TryComp(weapon, out weaponModulesComponent) || !TryComp(weapon, out appearanceComponent) ||
            ModulesSlot != args.Container.ID)
        {
            weaponModulesComponent = null;
            appearanceComponent = null;
            return false;
        }
        
        if(!weaponModulesComponent.Modules.Contains(module))
            weaponModulesComponent.Modules.Add(module);

        _appearanceSystem.SetData(weapon, ModuleVisualState.Module, component.AppearanceValue, appearanceComponent);
        
        return true;
    }
```
Аналогично с вытаскиванием. Но с вытаскиванием не нужно передавать в аргументы `BaseModuleComponent`, получается. По крайней мере пока у нас один модуль на оружие.